bin/util.py

```diff
@@ -391,18 +406,159 @@ def finalize(self, *, print_done=True, message=None):
+
+    def _print(self, *objects, sep='', end='\n', file=sys.stderr, flush=True):
+        assert self._is_locked()
+        # drop all flushes...
+        print(*objects, sep=sep, end=end, file=file, flush=False)
+
+    def start(self, item):
+        self.table.add_testcase(item.testcase.name)
+        return super().start(item)
+
+    def done(self, success=True, message='', data=''):
+        return super().done(success, message, data)
+
+    def finalize(self, *, print_done=True, message=None):
+        with self:
+            res = super().finalize(print_done=print_done, message=message)
+            self.table.clear(force=True)
+            return res
+
+
+class VerdictTable:
+    def __init__(
+        self, submissions, testcases, width=shutil.get_terminal_size().columns, max_name_width=50
+    ):
+        self.submissions = [
+            submission.name for verdict in submissions for submission in submissions[verdict]
+        ]
+        self.testcases = [testcase.name for testcase in testcases]
+        self.results = []
+        self.current_testcases = set()
+        self.name_width = min(
+            max_name_width, max([len(submission) for submission in self.submissions])
+        )
+        self.width = width if width >= self.name_width + 2 + 10 else -1
+        self.last_printed = []
+
+    def next_submission(self):
+        self.results.append(dict())
+        self.current_testcases = set()
+
+    def add_testcase(self, testcase):
+        self.current_testcases.add(testcase)
+
+    def finish_testcase(self, testcase, verdict):
+        self.results[-1][testcase] = verdict
+        self.current_testcases.discard(testcase)
+
+    def clear(self, *, force=True, clear=True):
+        if force or not config.args.no_bar:
+            if self.last_printed:
+                actual_width = shutil.get_terminal_size().columns
+                lines = 0
+                for printed in self.last_printed:
+                    lines += (printed + actual_width - 1) // actual_width
+                print(f'\033[{lines}A\r', end='', file=sys.stderr)
+
+                if clear:
+                    for printed in self.last_printed:
+                        print(' ' * printed, file=sys.stderr)
+                    print(f'\033[{lines}A\r', end='', file=sys.stderr)
+
+                self.last_printed = []
+
+    def _get_verdict(self, s, testcase):
+        if s < len(self.results):
+            if testcase in self.results[s]:
+                v = self.results[s][testcase]
+                if v == 'ACCEPTED':
+                    return Fore.GREEN + 'A' + Style.RESET_ALL
+                else:
+                    return Fore.RED + v[0] + Style.RESET_ALL
+            elif s + 1 == len(self.results) and testcase in self.current_testcases:
+                return Style.DIM + Fore.BLUE + '?' + Style.RESET_ALL
+        return Style.DIM + Fore.WHITE + '-' + Style.RESET_ALL
+
+    # TODO only print this if a flag is given! (config.args.???)
```

Comment:
I was going to propose `--table` as flag, but turns out we already have that :stuck_out_tongue: The only difference is that the existing `--table` flag has the test cases on the rows, instead of the submissions. Some options:
- Rename `--table` to `--verdict-analysis` (because that's what the existing table is used for), and use `--table` for this new Overview
- Keep `--table` what it is now, and think of a new name for this new Overview (like `--overview` or `--progress`)